lib/src/default_index/revset_engine.rs

```diff
@@ -829,6 +829,43 @@ impl<'index> EvaluationContext<'index> {
                     Ok(Box::new(EagerRevset { positions }))
                 }
             }
+            ResolvedExpression::ReachableWithin { sources, domain } => {
+                let mut sets = union_find::UnionFind::<IndexPosition>::new();
+
+                // Compute all reachable subgraphs.
+                let domain_revset = self.evaluate(domain)?;
+                let domain: HashSet<_> = domain_revset.positions().attach(index).collect();
```

Comment:
Instead of re-sorting `domain` at the end, it's probably better to collect into `Vec<>` here, and build a separate `HashSet<>`.